packages/core/src/errors.ts

```diff
@@ -1,13 +1,13 @@
 export class FileNotFoundError extends Error {
 	constructor(readonly filePath: string) {
-		super(`File not found: ${filePath}`);
+		super('File not found', { cause: { filePath } });
```

Comment:
I generally prefer not using non-errors as `cause` or in `throw`. Can we instead add a `filePath` property on the error, and then update the error-reporter to pick that property, and send it in `extras` ?